hare/eligibility/oracle.go

```diff
@@ -282,14 +283,14 @@ func (o *Oracle) prepareEligibilityCheck(ctx context.Context, layer types.LayerI
 		totalWeight *= uint64(committeeSize)
 		n *= committeeSize
 	}
-	p = fixed.DivUint64(uint64(committeeSize), totalWeight)
+	p := fixed.DivUint64(uint64(committeeSize), totalWeight)
 
-	if minerWeight > maxSupportedN {
+	if n > maxSupportedN {
```

Comment:
@noamnelke can you help review this part of code please? i want to make sure the math is correct.

Reply:
@noamnelke specifically I think that some of the casts and overflow checks (see comments) might be an issue.

Reply:
Created a new PR to specifically address these issues: https://github.com/spacemeshos/go-spacemesh/pull/4275